File: offset-manager/src/main/java/offsetmanager/exception/OffsetNotFoundException.java
package offsetmanager.exception;


import lombok.Getter;

@Getter
public class OffsetNotFoundException extends RuntimeException {
  private final String key;
  /**
   * Construct offsetmanager.exception avoiding stack trace for performance reasons.
   * @param key object key for which the offset was not found
   */
  public OffsetNotFoundException(String key) {
    super("Offset not found for key: " + key, null, false, false);
    this.key = key;
  }
}

File: offset-manager/src/main/kotlin/offsetmanager/exception/OffsetNotFoundException.kt
package offsetmanager.exception
/**
 * Construct offsetmanager.exception avoiding stack trace for performance reasons.
 * @param key object key for which the offset was not found
 */
class OffsetNotFoundException(
  private val key: String
) : RuntimeException(
  "Offset not found for key: $key", null,
  false, false
 )
